### DISCUSSION TITLE
What the hell is going on here? ;-)

### DISCUSSION THREAD

**Discussion by michaelbeijer** [goldendict/goldendict#1620] [General]
Can anyone here comment on what happened to the goldendict.org website (http://goldendict.org/), and the goldendict forum (http://goldendict.org/forum/)? They seem to have been frozen in time. This, and the fact that there have been no new binaries for Windows for years now, are the reason I started shopping around and stumbled across nonwill's GoldenDict-OCR and xiaoyifang's GoldenDict.ng. 

I have also read quite a few interesting comments here and there online indicating that not everyone in the GoldenDict community sees eye to eye, to put it mildly. On the one hand, this is a shame, as it threatens to put off new users. On the other hand, it's also perfectly possible to just choose one of the competing forks and use it and ignore all the crazy talk. 

However, I do think that a lot of damage is being done to GD's public image by the current sorry state of goldendict.org & goldendict.org/forum, which is the first place most newbies will end up upon discovering GoldenDict. They arrive at those two basically dead websites and might think that the project is dead, which is a shame, since it obviously isn't and lots of amazing development is going on on GitHub and elsewhere.

Michael

**Comment by nonwill** [comment]
The following may not answer your question, but
### here, just my onside suggestion:
All users who yearn for the WebEngine one of GoldenDict should switch to @vedgy's Qt WebEngine PR #1542. 

This is not an advertisement, nor is it a denigration of xiaoyifang's version.

### This is based on the following facts:
xiao's NGs are riddled with **massive memory usage clutter** and has **more buffer overflow vulnerabilities**, and has a lot of **logic errors** and **broke too many compatibilities**; Xiao's C/C++/Qt ability is averageand, but it likes to take credits for itself, itself has a small chicken belly and is **a willful executor**...
### For normal users, your experience is:
NG starts too **slowly** and runs too **slowly**, and various **problems** when displaying content, sometime not running as the official one; NG takes up **_too much memory_** and occupies **_too much CPU_**, and **_often crashes randomly_**...
### But xiao would answer you as:
This is a Qt problem, or it is a problem with WebEngine, even or it is a problem with dictionaries... I think it's... I think it might be... I'm not sure😁...
**Scariest things are that xiao would say**: 
Those codes are fine (and then it will quietly change them at some point in the future)😁...

It is not maliciously smearing xiao and its ng, but they all do behave as these. 
Though I think there are lot of positives to the birth of xiao‘s NG, I don't think the NG is a useful one, it's just a low-quality replica of two people involved.

**If some WebEngine one is the future of GoldenDict, as far as we can see now it must come from @vedgy's Qt WebEngine PR #1542.**

**Reply by vedgy** [reply]
@nonwill, the person who started this discussion appears to be primarily interested in Windows builds. He may have little need for Qt WebEngine. And I wouldn't expect him to go to the trouble of compiling the #1542 version. Besides, my version may work poorly on Windows, because I (and possibly no one) have never tested it on this platform. And there are two Windows-specific Qt WebEngine TODOs that I personally cannot address, because I don't volunteer development on Windows.

**Reply by michaelbeijer** [reply]
Indeed, I am on Windows, and don't know anything about Qt WebEngine or Linux. I just want the latest versions of GD that works well on my massive collection of dictionaries, so I can use GD in my work as a technical/patent translator. I also want binaries so I can just install it and use it without messing around too much. 

Also, ideally, I would not have to use Google Translate to translate either Chinese or Russian just to figure out where to find the program and read about any new features ;-)

**Reply by michaelbeijer** [reply]
Since, to an English speaker, the easiest to use is xiaoyifang's GoldenDict.ng, that's the main one I am testing currently. It seems to work very well. For these reasons this is also the version I am currently recommending to my translator colleagues/friends. 

However, nonwill's version seems to be equally good in terms of the speed of indexing and new features, and I am also following its development closely.

**Reply by nonwill** [reply]
> I wouldn't expect him to go to the trouble of compiling the #1542 version.

Another version in https://github.com/goldendict/goldendict/issues/1590#issuecomment-1474103429 was compiled based on #1542, which works on Windows, but it still had some problems which should be caused by the confusion of third-party libraries, which were mixed with the officials and GoldenDict++'s, some of them may also be not compatible with those built with Qt-5.15.x.

Though I still love the lightness of the WebKit version, which is fast and doesn't take up much disk spaces, but there should be no serious issues blocking #1542 to be built and running on Windows.

**Reply by shenlebantongying** [reply]
Nonwill is a middle age loser trying to boost his ego and finally find some purpose of life by sabotaging this tiny program.

The fact that he has his own fork, and now he is promoting vedgy's unlikly-to-be-merged PR is just holy fucking shit level of sneaky.

**Reply by nonwill** [reply]
It's fair and aboveboard, please don't wrong yourself as you just doing，🐷brothers🐷.

**Comment by vedgy** [comment]
> This, and the fact that there have been no new binaries for Windows for years now, are the reason I started shopping around and stumbled across nonwill's GoldenDict-OCR and xiaoyifang's GoldenDict.ng.

The single remaining maintainer of this repository regularly posts fresh Windows executables [on a Russian forum](https://forum.ru-board.com/topic.cgi?forum=5&topic=49149&start=0). The maintainer has decided to stop updating early access build pages here for some reason. I suspect that the fact that non-contributors have edited those pages further put him off. There has even been recent vandalism (replacing links with tracking redirects) on [the Windows Early Access Builds wiki page](https://github.com/goldendict/goldendict/wiki/Early-Access-Builds-for-Windows). Fortunately, after @wildekat studiously took on reverting the malicious link replacements, the vandalism stopped half a year go.

> I have also read quite a few interesting comments here and there online indicating that not everyone in the GoldenDict community sees eye to eye, to put it mildly. 

The differences in opinion are drastic and there is little prospect of reaching common ground. Therefore, I hope everyone can focus on participation in the repository they prefer and avoid interacting with the competing one. This way future conflicts can be prevented.

> However, I do think that a lot of damage is being done to GD's public image by the current sorry state of goldendict.org & goldendict.org/forum, which is the first place most newbies will end up upon discovering GoldenDict.

For some reason, the author of the desktop and mobile GoldenDict version, as well as all but one maintainer, stopped participating in GoldenDict development almost 10 years ago. The single remaining GoldenDict maintainer likely has neither permission nor motivation to maintain the website and the forum.

**Reply by cjc7373** [reply]
> The single remaining maintainer of this repository regularly posts fresh Windows executables [on a Russian forum](https://forum.ru-board.com/topic.cgi?forum=5&topic=49149&start=0).

Perhaps we should point this out in the wiki page, as users may want to get the latest builds.

**Reply by vedgy** [reply]
Good idea. Just updated https://github.com/goldendict/goldendict/wiki/Early-Access-Builds-for-Windows

**Reply by michaelbeijer** [reply]
Thanks for the link, and extra info on where to find the file (the link after "последний EXE-файл: ")!

**Comment by shenlebantongying** [comment]
What's happening here is that no serious PR getting merged.

A cancer is trying to sabotage this project.

vedgy don't want real improvements from happening and keep trying to keep others from helping out.

This project is beyond dead. Both people mentioned above have mental illness.

Thankfully, both of them are not doing anything. Ignoring them is OK.

**Reply by vedgy** [reply]
> What's happening here is that no serious PR getting merged.
> 
> Thankfully, both of them are not doing anything. Ignoring them is OK.

That's not true as anyone can see for themselves here: https://github.com/goldendict/goldendict/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Aclosed. Most pull requests are merged. The merged PRs contain code cleanups, bug fixes and new features.

> vedgy don't want real improvements from happening and keep trying to keep others from helping out.
> 
> This project is beyond dead. Both people mentioned above have mental illness.

@shenlebantongying, I don't know what you are referring to, but you definitely do not "help out" the GoldenDict project. I review others' pull requests in good faith. @Abs62 has merged many of them after my thorough review and approval. Especially GNU/Linux-specific pull requests.

**Reply by shenlebantongying** [reply]
You are now Collaborator, Just merge your PR and finally have a ""OFFICIAL"" gd that works on modern systems.

Just find someone to review it and click the merge button.

> That's not true as anyone can see for themselves here: https://github.com/goldendict/goldendict/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Aclosed. Most pull requests are merged. The merged PRs contain code cleanups, bug fixes and new features.

Holy molly. The last two merged PR are just fixing deprecations, and the real one from 3 mouths ago. 1 PR per 3 mouths?? 2 PR in total in the past half of years?????

**Reply by vedgy** [reply]
> Holy molly. The last two merged PR are just fixing deprecations, and the real one from 3 mouths ago. 1 PR per 3 mouths?? 2 PR in total in the past half of years?????

GoldenDict contributors are volunteers who work in their free time. I prefer slow and steady development of the mature and stable application over introducing numerous regressions by incompetent and careless coding.

**Reply by shenlebantongying** [reply]
> incompetent

You are indeed a competent programmer. Wow, how amazing is that. Just think about it, always do right multicore programming and knows everything about c++.

Ha, should incompetent programmers be banned from participating? What's your bar of participating? A PhD in CS? For a little dictionary program???

So, this is a project led by elite programmers like you. Truly my bad for not knowing that.
